### PR TITLE
Add default MFA device selection for OneLogin

### PR DESCRIPTION
**Pull request by brandonstrohmeyer**
This PR adds support for a default MFA device with OneLogin, saving the user the trouble of the single keystroke to select their MFA device from a list. 

There are two primary features:
1. A new `--mfa-device` flag and global option. This flag matches the device name returned from OneLogin. 
2. YubiKey Autodetection. If a YubiKey is connected to the machine executing Clisso, it will automatically be selected as the MFA device to use.

The YubiKey autodetection relies on the karalabe/hid package which needs CGO enabled and appropriate C toolchain installed in order to cross compile. I did not make these changes to .goreleaser as I'm not sure exactly how the build machine is setup and what toolchains are available to use. I'm happy to make any requested changes, but am asking for a little guidance here. Opening as a Draft until the .goreleaser issue is resolved.

### CODE REVIEW

**Comment by bitte-ein-bit** [comment]
Hey @brandonstrohmeyer,

thank you for this PR! Please don't forget to accept the Contributer License Agreement.

Goreleaser is a little tricky when it comes to CGO. Bytebase did a pretty awesome [walk through](https://www.bytebase.com/blog/how-to-cross-compile-with-cgo-use-goreleaser-and-github-action/). I fear we have to go a similar approach.

Unfortunately, that also means I have to move the MacOS notarization to Github actions. That's going to be a little bigger pain.

If you would be so kind to follow the bytebase write-up, then I can take care of the notarization later.

Thanks

**Comment by bitte-ein-bit** [comment]
@brandonstrohmeyer  For the time being, may I suggest to split the two (Yubikey detection, flag for default MFA)?

**Comment by bitte-ein-bit** [comment]
@brandonstrohmeyer I've reworked the build system. Can you test that I didn't mess up while merging the changes from the master branch? I don't own a Yubikey to test.

Please also sign the [Contributor Agreement](https://cla-assistant.io/allcloud-io/clisso?pullRequest=367).

**Comment by brandonstrohmeyer** [comment]
@bitte-ein-bit apologies for going MIA, things got busy and side projects fell by the wayside...It really doesn't feel like I opened this PR nearly a year ago.

I have signed the CLA.

I have tested all the changes you've made, and after building the macos binary locally and signing it with a self-signed cert everything was still working as expected.

I see your previous comment:

> For the time being, may I suggest to split the two (Yubikey detection, flag for default MFA)?

Do you still want me to move yubikey detection to its own PR? No trouble if so, I'm just trying to get reacquainted with this project.

File: onelogin/get.go [hunk 1]
```diff
@@ -19,15 +19,20 @@ import (
 	"github.com/allcloud-io/clisso/log"
 	"github.com/allcloud-io/clisso/saml"
 	"github.com/allcloud-io/clisso/spinner"
+	"github.com/allcloud-io/clisso/yubikey"
 	"github.com/icza/gog"
 	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
```
Comment (bitte-ein-bit) [inline onelogin/get.go:25]: When fixing conflicts, please note that `logrus` was encapsulated in our own log package.

File: onelogin/get.go [hunk 2]
```diff
@@ -253,6 +304,28 @@ func getDevice(devices []Device) (device *Device, err error) {
 		return
 	}
 
+	if opts.MfaDevice != "" {
+		for _, d := range devices {
+			if d.DeviceType == opts.MfaDevice {
+				device = &d
+				fmt.Printf("MFA device %s found, automatically selecting it.\n", opts.MfaDevice)
```
Comment (bitte-ein-bit) [inline onelogin/get.go:311]: These print statements should be adopted to the new logging mechanism.

File: yubikey/yubikey.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package yubikey
+
+import (
+	"github.com/karalabe/hid"
+	log "github.com/sirupsen/logrus"
+)
+
+// IsAttached queries the connected USB devices and returns true if a YubiKey is attached
+func IsAttached() bool {
+	var yubiKeyVendorID uint16 = 0x1050
```
Comment (bitte-ein-bit) [inline yubikey/yubikey.go:10]: that should be some constant I guess

File: onelogin/get.go [hunk 4]
```diff
@@ -253,6 +304,28 @@ func getDevice(devices []Device) (device *Device, err error) {
...
+				return
+			}
+		}
+		// If the user requested device is not found, fall through and continue the device selection process.
+		fmt.Printf("MFA device %s not found.\n", opts.MfaDevice)
+	}
+
+	if opts.AutodetectYubiKey {
+		for _, d := range devices {
+			if d.DeviceType == MFADeviceYubicoYubiKey {
```
Comment (bitte-ein-bit) [inline onelogin/get.go:321]: Out of curiosity, if you have more than two yubikeys plugged in, the selection will be random. I mean, why would you. ;-)
But maybe worth noting in the Readme as a limitation.

**Comment by bitte-ein-bit** [comment]
Hey @brandonstrohmeyer,

Yeah, time flies very fast. 

No, there is no reason to split the two items anymore. I wanted to move forward faster without completely reworking the build system. But that's out of the way now, so we should be good. Unfortunately, I couldn't fully get CGO setup on some OS/Archs combinations. Major x86 versions should be covered. Please add some remarks in the readme.

Let's fix the conflicts, get it tested and merged.

I've left some comments in the code.

File: onelogin/get.go (hunk 4 above)
Comment (brandonstrohmeyer) [reply onelogin/get.go:328]: Noted in the readme

File: yubikey/yubikey.go (hunk 3 above)
Comment (brandonstrohmeyer) [reply yubikey/yubikey.go]: Moved to constant

File: onelogin/get.go (hunk 2 above)
Comment (brandonstrohmeyer) [reply onelogin/get.go]: Moved to log statements

**Comment by bitte-ein-bit** [comment]
@brandonstrohmeyer Snyk lists some dependency chain issues. But none with a fix available.

**Pull request merged**